### ISSUE TITLE
Ads do not refresh on route change

### ISSUE THREAD

**Issue by Soviut** [ianmurrays/ngDfp#3]
I have a big box ad on each sub page of my site. The ad displays properly on initial load but on route changes the ad won't reload and the area remains blank. My workaround is to call `refreshAds()` on `$routeChangeSuccess` events.

``` javascript
    $rootScope.$on '$routeChangeSuccess', (e, newRoute, oldRoute) ->
        DoubleClick.refreshAds 'div-gpt-ad-1234567890'
```

**Comment by guyyosan** [comment]
@Soviut I guess that's coffee script
Could you maybe explain how would you do it in pure js using ui-router instead?

**Comment by ianmurrays** [comment]
@guyyosan that would translate to 

```javascript
$rootScope.$on('$stateChangeStart', function () {
  DoubleClick
    .getSlot('div-gpt-ad-1234567890')
    .then(function (slot) {
      googletag.pubads().clear([slot]);
    });
});
```

and

```javascript
$rootScope.$on('$stateChangeSuccess', function () {
  $timeout(function () {
    DoubleClick.refreshAds('div-gpt-ad-1234567890');
  }, 2000);
});
```

**Comment by guyyosan** [comment]
@ianmurrays does it have to be in the $rootScope?
and if it must, where do I put it... controller or run?

**Comment by ianmurrays** [comment]
Yes I believe it has to observe the `$rootScope`, and the `.run()` call makes more sense to me at least.

**Comment by guyyosan** [comment]
@ianmurrays It's a dumb question, guess im dumb... :(
what's the slot?
                                                    this?                     or                  this?
                .defineSlot('/123456/blogbox11233', [[300, 250]], 'homePage1')

**Comment by ianmurrays** [comment]
in your example `homePage1` would be the slot

**Comment by guyyosan** [comment]
@ianmurrays 
First thanks a lot for this module it saved my life.
second I erased this from the module:
    var googletag     = googletag || {};
    googletag.cmd = googletag.cmd || [];

and inserted it in my app.js like this:
(function() {
    'use strict';

    var googletag     = googletag || {};
    googletag.cmd = googletag.cmd || [];
var app = angular

secondly, If I want multiple ads would this be alright:

rootScope.$on('$stateChangeStart', function(){
                    $rootScope.state_loading = true;
                    DoubleClick
                        .getSlot('homePage1')
                        .then(function (slot) {
                            googletag.pubads().clear([slot]);
                        })
                        .getSlot('homePage2')
                        .then(function (slot) {
                            googletag.pubads().clear([slot]);
                        })
                        .getSlot('homePage3')
                        .then(function (slot) {
                            googletag.pubads().clear([slot]);
                        });
                });

                $rootScope.$on('$stateChangeSuccess', function(){
                    $rootScope.state_loading = false;

                    $timeout(function() {
                        DoubleClick.refreshAds('homePage1', 'homePage2', 'homePage3');
                    }, 2000);
                });

**Comment by guyyosan** [comment]
@ianmurrays this is what i get:


n.getSlot(...).then(...).getSlot is not a function
@http://localhost/dist/bundle-ctrl.js:1:4187
lf/this.$get</n.prototype.$broadcast@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:136:43
b/E.transitionTo@http://localhost/dist/bundle-mdl.js:1:31202
b/E.go@http://localhost/dist/bundle-mdl.js:1:29817
P/<.link/</i<@http://localhost/dist/bundle-mdl.js:2:6255
f/m<@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:146:220
e@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:43:93
Kf/l.defer/c<@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:45:491
 angular.min.js:107:207

Error: e.pubads is not a function
@http://localhost/dist/bundle-ctrl.js:1:4227
f/<@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:119:113
lf/this.$get</n.prototype.$eval@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:133:217
lf/this.$get</n.prototype.$digest@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:130:225
lf/this.$get</n.prototype.$apply@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:133:510
f/m<@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:146:298
e@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:43:93
Kf/l.defer/c<@https://ajax.googleapis.com/ajax/libs/angularjs/1.4.7/angular.min.js:45:491

**Comment by ianmurrays** [comment]
`getSlot()` isn't chainable. You might have luck dividing those calls into multiple.
